### PR TITLE
More prototype Transforms cleanups

### PR DESCRIPTION
**Pull request by datumbox**
This PR:
- Makes Normalize verify input on forward instead on the kernel
- The kernel `to_image_tensor` returns `features.Image` instead of plain tensor.
- The kernel `decode_image_with_pil` returns `features.Image` instead of plain tensor.

### CODE REVIEW

**Review by vfdev-5 — APPROVED** [review]
LGTM, thanks

**Review by pmeier — APPROVED** [review]
LGTM, thanks!

**Review by vfdev-5 — APPROVED** [review]
Still good to me

File: torchvision/prototype/features/_encoded.py [hunk 1]
```diff
@@ -46,11 +46,6 @@ def image_size(self) -> Tuple[int, int]:
 
         return self._image_size
 
-    def decode(self) -> Image:
```
Comment (datumbox) [inline torchvision/prototype/features/_encoded.py:49]: Mypy was complaining with:
```
torchvision/prototype/features/_encoded.py:52: error: Returning Any from function declared to return "Image"  [no-any-return]
            return self._F.decode_image_with_pil(self)
```

It's unclear why it couldn't see that the return type is indeed `Image` but given it's not used anywhere in the codebase and it's for debugging only, I'm removing it. Feel free to add back on the future,

File: torchvision/prototype/transforms/_misc.py [hunk 2]
```diff
@@ -104,6 +104,11 @@ def __init__(self, mean: Sequence[float], std: Sequence[float]):
     def _transform(self, inpt: Any, params: Dict[str, Any]) -> Any:
         return F.normalize(inpt, mean=self.mean, std=self.std)
 
+    def forward(self, *inpts: Any) -> Any:
+        if has_any(inpts, PIL.Image.Image):
+            raise TypeError(f"{type(self).__name__}() does not support PIL images.")
+        return super().forward(*inpts)
```
Comment (datumbox) [inline torchvision/prototype/transforms/_misc.py:110]: Adopting @vfdev-5's proposal to eliminate the `sample = inputs if len(inputs) > 1 else inputs[0]` idiom in favour of `super().forward(*inpts)`
